views/js/pciCreator/dev/textReaderInteraction/runtime/textReaderInteraction.amd.js

```diff
@@ -0,0 +1,168 @@
+/*global define,$*/
+define(
+    [
+        'qtiCustomInteractionContext',
+        'OAT/util/event',
+        'textReaderInteraction/runtime/js/renderer'
+    ],
+    function (qtiCustomInteractionContext, event, Renderer) {
+        'use strict';
+        qtiCustomInteractionContext.register({
+            id : -1,
+            getTypeIdentifier : function () {
+                return 'textReaderInteraction';
+            },
+            /**
+             * Render the PCI : 
+             * @param {String} id
+             * @param {Node} dom
+             * @param {Object} config - json
+             */
+            initialize : function (id, dom, config) {
+                var that = this,
+                    pci = this._taoCustomInteraction,
+                    properties = that._taoCustomInteraction.properties,
+                    pageIds = _.pluck(properties.pages, 'id'),
+                    maxPageId = Math.max.apply(null, pageIds);
+
+                this.id = id;
+                this.dom = dom;
+                this.config = config || {};
+                this.$container = $(dom);
+
+                if (!pci.widgetRenderer) {//delivery context
+                    pci.widgetRenderer = new Renderer({
+                        serial : pci.serial,
+                        $container : this.$container
+                    });
+                    pci.widgetRenderer.renderAll(pci.properties);
+                }
+
+                //add method on(), off() and trigger() to the current object
+                event.addEventMgr(this);
+                //tell the rendering engine that I am ready
+                qtiCustomInteractionContext.notifyReady(this);
+
+                this.$container.on('click', '.js-add-choice', function () {
```

Comment:
same issue as the previous comment  : please move this code to the authoring widget.